File: algorithms/solutions/java/_0128_LongestConsecutiveSequence.java
package hot100;

import java.util.Arrays;
import java.util.Set;
import java.util.stream.Collectors;

public class _0128_LongestConsecutiveSequence {
// @lc code=start
    class Solution {
        public int longestConsecutive(int[] nums) {
            if (nums.length == 0) return 0;
            Set<Integer> set = Arrays.stream(nums).boxed().collect(Collectors.toSet());
            int maxLen = 1;
            for(int num : nums){
                if (!set.contains(num - 1)){
                    int curNum = num;
                    int curLen = 1;
                    while(set.contains(curNum + 1)){
                        curNum++;
                        curLen++;
                    }
                    maxLen = Math.max(maxLen, curLen);
                }
            }
            return maxLen;
        }
    }
// @lc code=end
}

File: algorithms/solutions/python3/0128_longest_consecutive_sequence.py
#
# @lc app=leetcode.cn id=128 lang=python3
#
# [128] Longest Consecutive Sequence
#
# https://leetcode.cn/problems/longest-consecutive-sequence/description/

# @lc code=start
from typing import List


class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0
        max_len = 1
        s = set(nums)
        for num in nums:
            if (num - 1) not in s:
                cur_len = 1
                cur_num = num
                while (cur_num + 1) in s:
                    cur_len += 1
                    cur_num += 1
                max_len = max(cur_len, max_len)
        return max_len
                    
            
        
# @lc code=end

